### ISSUE TITLE
Document methods and structs

### ISSUE THREAD

**Issue by Stupremee** [Kixiron/lsw#1]
It would be nice to provide a short and good explanation to everything, and also a url to the LLVM counterpart.
Something like this:
```
A Module instance is used to store all the information related to an LLVM module. 

See [here](http://llvm.org/doxygen/classllvm_1_1Module.html#details) for details.
```

It's much easier if you start from the beginning writing docs, because doing it afterwards is really annoying.
We could also add some lint about missing docs or probably even `deny` it.

**Comment by Kixiron** [comment]
I totally agree, which is why I've been mostly trying to do them as I go. The only problem is that it can be difficult to provide examples when there's required things that aren't yet implemented, but we can always go back
